tests/khalendar_test.py

```diff
@@ -339,9 +339,10 @@ def test_invalid_timezones(self, coll_vdirs):
         assert events[0].format('{start} {end} {title}', dt.date.today()) == \
             '02.12. 08:00 02.12. 09:30 Some event\x1b[0m'
 
-    def test_multi_uid_vdir(self, coll_vdirs, caplog, fix_caplog):
+    def test_multi_uid_vdir(self, coll_vdirs, caplog, fix_caplog, sleep_time):
         coll, vdirs = coll_vdirs
         caplog.set_level(logging.WARNING)
+        sleep(sleep_time)
```

Comment:
Are these required due to poor clock resolution on the CI environment or something?

Reply:
I'm not sure about the details, but when I traced the failures I found that `needs_update()` was false, and on further investigation the ctag in the DB and the filesystem were returning the [same timestamp](https://github.com/pimutils/khal/runs/3156981790?check_suite_focus=true#step:6:183) (to what *looks* like high resolution).

I assume that the DB ctag is being created/reset at the start of the test, and if the upload step is faster than the clock resolution the test fails. The sleep used elsewhere in `khalendar_test.py` (~0.011s) seems to be enough to avoid this.

I guess I should comment these too...

Reply:
Note that they all end in `.000000000`, so it seems they're not high resolution. I wonder what filesystem is being used under the hood that has this issue. AFAIK, we've never seen anything this low resolution.

Some background:

`needs_update` will only return `True` if the file has changed since we last read it. If the clock is **too** low precision, then when we edit the file a second time, it's mtime hasn't changed (since the clock hasn't moved forward due to its lack of precision).

In simpler terms: if the clock only has milisecond precision, then we must wait at least 1.1 miliseconds between edits for the timestamp to change. `sleep_time` determines how much time we need to sleep (see its docs for details).